### PR TITLE
ci: perf test for master

### PR DESCRIPTION
**Pull request by KateKate**
:baby_chick: Download canary assets:

[colors--canary.402.4383968486.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.402.4383968486.xml)  
[color_metro_ios-swift--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.402.4383968486.swift)  
[color_metro_kotlin--canary.402.4383968486.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.402.4383968486.kt)  
[color_metro_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.402.4383968486.ts)  
[color_sberHealth_ios-swift--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.402.4383968486.swift)  
[color_sberHealth_kotlin--canary.402.4383968486.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.402.4383968486.kt)  
[color_sberHealth_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.402.4383968486.ts)  
[color_sbermarket_ios-swift--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.402.4383968486.swift)  
[color_sbermarket_kotlin--canary.402.4383968486.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.402.4383968486.kt)  
[color_sbermarket_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.402.4383968486.ts)  
[color_sberprime_ios-swift--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.402.4383968486.swift)  
[color_sberprime_kotlin--canary.402.4383968486.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.402.4383968486.kt)  
[color_sberprime_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.402.4383968486.ts)  
[color_selgros_ios-swift--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.402.4383968486.swift)  
[color_selgros_kotlin--canary.402.4383968486.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.402.4383968486.kt)  
[color_selgros_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.402.4383968486.ts)  
[color_smbusiness_ios-swift--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.402.4383968486.swift)  
[color_smbusiness_kotlin--canary.402.4383968486.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.402.4383968486.kt)  
[color_smbusiness_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.402.4383968486.ts)  
[PlasmaTokensColor--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.402.4383968486.swift)  
[shadow_sbermarket_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.402.4383968486.ts)  
[typo_mage_ios-swift--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.402.4383968486.swift)  
[typo_mage_kotlin--canary.402.4383968486.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.402.4383968486.kt)  
[typo_mage_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.402.4383968486.ts)  
[typo_plasma_ios-swift--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.402.4383968486.swift)  
[typo_plasma_kotlin--canary.402.4383968486.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.402.4383968486.kt)  
[typo_plasma_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.402.4383968486.ts)  
[typo_ruler_ios-swift--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.402.4383968486.swift)  
[typo_ruler_kotlin--canary.402.4383968486.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.402.4383968486.kt)  
[typo_ruler_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.402.4383968486.ts)  
[typo_sage_ios-swift--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.402.4383968486.swift)  
[typo_sage_kotlin--canary.402.4383968486.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.402.4383968486.kt)  
[typo_sage_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.402.4383968486.ts)  
[typo_sbermarket_ios-swift--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.402.4383968486.swift)  
[typo_sbermarket_kotlin--canary.402.4383968486.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.402.4383968486.kt)  
[typo_sbermarket_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.402.4383968486.ts)  
[typo_soulmate_ios-swift--canary.402.4383968486.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.402.4383968486.swift)  
[typo_soulmate_kotlin--canary.402.4383968486.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.402.4383968486.kt)  
[typo_soulmate_react-native--canary.402.4383968486.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.402.4383968486.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-temple@1.140.2-canary.402.4383968486.0
  npm install @salutejs/plasma-ui@1.173.2-canary.402.4383968486.0
  # or 
  yarn add @salutejs/plasma-temple@1.140.2-canary.402.4383968486.0
  yarn add @salutejs/plasma-ui@1.173.2-canary.402.4383968486.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-402/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-402/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-402/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-402/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-402/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-402/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-402/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-402/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-402/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-402/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-402/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-402/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-402/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-402/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-402/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-402/temple-storybook/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-402/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-402/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-402/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-402/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-402/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-402/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-402/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-402/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-402/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-402/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-402/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-402/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-402/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-402/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-402/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-402/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-402/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-402/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-402/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-402/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-402/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-402/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-402/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

File: .github/workflows/performance-test-pr.yml [hunk 1]
```diff
@@ -15,98 +15,99 @@ jobs:
...
+          key: npm-deps-${{ hashFiles('package-lock.json') }}
+          restore-keys: |
+            npm-deps-${{ hashFiles('package-lock.json') }}
+
+      - name: Setup packages for PR
+        run: |
+          npm ci
+          npx lerna bootstrap
+
+      - name: Use Node.js 16.15.x for PR
```
Comment (akhdrv) [inline .github/workflows/performance-test-pr.yml:77]: мб один раз достаточно выполнить этот шаг?

Comment (akhdrv) [reply .github/workflows/performance-test-pr.yml:77]: а, вижу, ты меняешь на 14 и обратно

File: .github/workflows/performance-test-pr.yml [hunk 2]
```diff
@@ -15,98 +15,99 @@ jobs:
...
 
-      - name: Save pr report result
-        if: steps.mainReportCheck.outputs.files_exists == 'false' && always()
-        uses: actions/upload-artifact@v2
-        with:
-          name: Component performance comparison result
-          path: /home/runner/work/plasma/plasma/packages/plasma-ui/perftest/result.json
-
       - name: Post result comment
-        if: always()
```
Comment (akhdrv) [inline .github/workflows/performance-test-pr.yml:96]: давай вернем, чтобы коммент писался всегда. в целевом решении джоба падает, если деграднули. и надо будет явно писать, что упали не потому что сломались

File: .github/workflows/performance-test-pr.yml [hunk 3]
```diff
@@ -15,98 +15,99 @@ jobs:
...
             let result = null;
             try {
-              report = hasMainReport ? require('/home/runner/work/plasma/plasma/packages/plasma-ui/perftest/comparison.json') : null;
+              report = require('/home/runner/work/plasma/plasma/packages/plasma-ui/perftest/comparison.json');
             } catch (e) {}
-            if (!hasMainReport) {
-              result = '🌚 Missing base report';
-            } else if (!report) {
+            if (!report) {
               result = '💀 Missing report, something went wrong';
```
Comment (akhdrv) [inline .github/workflows/performance-test-pr.yml:111]: думаю что можно не обрабатывать такие ошибки (чего спамить в PR если джоба сломалась). try-catch на require и проверку на report можно убрать

**Review by akhdrv — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-402/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-402/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-402/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-402/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-402/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-402/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-402/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Review by neretin-trike — APPROVED** [review]

**Pull request merged**